sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -335,13 +366,16 @@ class Analyzer(
 
         Aggregate(groupingAttrs, aggregations, expand)
 
-      case f @ Filter(cond, child) if hasGroupingFunction(cond) =>
+      // We should make sure all expressions in condition have been resolved.
```

Comment:
expression in condition could be unresolved, we should check this and respect the operator until `Filter.condition` is resolved. A testcase in `ResolveGroupingAnalyticsSuite` would fail before this PR.